tests/Unit/DataStructures/VectorImplTestHelper.hpp

```diff
@@ -161,5 +161,358 @@ auto remove_nth(const std::tuple<ValueTypes...>& tup) noexcept {
+                          make_not_null(&dist));
+  VectorType ref_generated_vector;
+  ref_generated_vector.set_data_ref(make_not_null(&generated_vector));
+  VectorType larger_generated_vector{size + 1, 0};
+  fill_with_random_values(make_not_null(&larger_generated_vector),
+                          make_not_null(&gen), make_not_null(&dist));
+  // this line should error, the reference should have received the smaller size
+  ref_generated_vector = larger_generated_vector;
+}
+
+/// \ingroup TestingFrameworkGroup
+/// \brief test that reference to a `VectorType` appropriately errors out when
+/// moved to another `VectorType` with the wrong size that is non-owning.
+///
+/// \details a calling function should be an `ASSERTION_TEST()` and check for
+/// the string "Must copy into same size".
+template <typename VectorType,
+          typename ValType = typename VectorType::ElementType>
+void vector_ref_test_move_size_error(
+    typename tt::get_fundamental_type_t<ValType> low =
+        typename tt::get_fundamental_type_t<ValType>{-100},
+    typename tt::get_fundamental_type_t<ValType> high =
+        typename tt::get_fundamental_type_t<ValType>{100}) noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<typename tt::get_fundamental_type_t<ValType>> dist{
+      low, high};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t size = sdist(gen);
+  VectorType generated_vector{size, 0};
+  fill_with_random_values(make_not_null(&generated_vector), make_not_null(&gen),
+                          make_not_null(&dist));
+  VectorType ref_generated_vector;
+  ref_generated_vector.set_data_ref(make_not_null(&generated_vector));
+  VectorType larger_generated_vector{size + 1, 0};
+  fill_with_random_values(make_not_null(&larger_generated_vector),
+                          make_not_null(&gen), make_not_null(&dist));
+  // this line should error, the reference should have received the smaller size
+  ref_generated_vector = std::move(larger_generated_vector);
+}
+
+/// \ingroup TestingFrameworkGroup
+/// \brief tests a small sample of math functions after movement of a
+/// `VectorType` initialized with `ValType`
+template <typename VectorType, typename ValType>
+void vector_test_math_after_move(
+    typename tt::get_fundamental_type_t<ValType> low =
+        typename tt::get_fundamental_type_t<ValType>{-100},
+    typename tt::get_fundamental_type_t<ValType> high =
+        typename tt::get_fundamental_type_t<ValType>{100}) noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<typename tt::get_fundamental_type_t<ValType>> dist{
+      low, high};
+  UniformCustomDistribution<size_t> sdist{2, 20};
+
+  size_t size = sdist(gen);
+  auto generated_val1 = make_with_random_values<ValType>(make_not_null(&gen),
+                                                         make_not_null(&dist)),
+       generated_val2 = make_with_random_values<ValType>(make_not_null(&gen),
```

Comment:
We prefer single initialization per statement, so `const auto generated_val2 = `